azurerm/resource_arm_data_factory.go

```diff
@@ -0,0 +1,371 @@
+		if err := d.Set("github_configuration", repo); err != nil {
+			return fmt.Errorf("Error setting Data Factory %q `github_configuration` (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if repoType == datafactory.TypeFactoryRepoConfiguration {
+		d.Set("vsts_configuration", repo)
+		d.Set("github_configuration", repo)
+	}
+
+	if err := d.Set("identity", flattenArmDataFactoryIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("Error setting Data Factory %q `identity` (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDataFactoryDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["factories"]
+
+	response, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("Error deleting Data Factory %s: %+v", name, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmDataFactoryRepoConfiguration(d *schema.ResourceData) (bool, datafactory.BasicFactoryRepoConfiguration) {
+	if vstsList, ok := d.GetOk("vsts_configuration"); ok {
+		vsts := vstsList.([]interface{})[0].(map[string]interface{})
+		accountName := vsts["account_name"].(string)
+		branchName := vsts["branch_name"].(string)
+		projectName := vsts["project_name"].(string)
+		repositoryName := vsts["repository_name"].(string)
+		rootFolder := vsts["root_folder"].(string)
+		tenantID := vsts["tenant_id"].(string)
+		// https://github.com/Azure/go-autorest/issues/307
+		return true, &datafactory.FactoryVSTSConfiguration{
+			AccountName:         &accountName,
+			CollaborationBranch: &branchName,
+			ProjectName:         &projectName,
+			RepositoryName:      &repositoryName,
+			RootFolder:          &rootFolder,
+			TenantID:            &tenantID,
+		}
+	}
+
+	if githubList, ok := d.GetOk("github_configuration"); ok {
+		github := githubList.([]interface{})[0].(map[string]interface{})
+		accountName := github["account_name"].(string)
+		branchName := github["branch_name"].(string)
+		gitURL := github["git_url"].(string)
+		repositoryName := github["repository_name"].(string)
+		rootFolder := github["root_folder"].(string)
+		// https://github.com/Azure/go-autorest/issues/307
+		return true, &datafactory.FactoryGitHubConfiguration{
+			AccountName:         &accountName,
+			CollaborationBranch: &branchName,
+			HostName:            &gitURL,
+			RepositoryName:      &repositoryName,
+			RootFolder:          &rootFolder,
+		}
+	}
+
+	return false, nil
+}
+
+func flattenArmDataFactoryRepoConfiguration(factory *datafactory.Factory) (datafactory.TypeBasicFactoryRepoConfiguration, []interface{}) {
+	result := make([]interface{}, 0)
+	properties := factory.FactoryProperties
+	if properties != nil {
```

Comment:
heads-up these can be merged into one:

```suggestion
	if properties := factory.FactoryProperties; properties != nil {
```